src/background/listener/bee-api.listener.ts

```diff
@@ -0,0 +1,85 @@
+import { StoreObserver, getItem } from '../../utils/storage'
+export class BeeApiListener {
+  private beeApiUrl: string
+
+  public constructor(private storeObserver: StoreObserver) {
+    this.beeApiUrl = 'http://localhost:1633'
+    this.addStoreListeners()
+    this.asyncInit()
+    this.addBzzListeners()
+  }
+
+  private addBzzListeners() {
+    // typed in 'bzz://{content-address}' URI in search bar triggers redirect to gateway BZZ address
+    // only react if google search has been set for searching
+    chrome.webRequest.onBeforeRequest.addListener(
```

Comment:
Not too familiar with chrome extensions but usually listeners should be removed somewhere otherwise this screams memory leaks.

Reply:
Right, but there surely is some 'on extension disabled' event when user manually disables the extension or uninstalls it. Though I could not find this in the docs at first glance (can look more tomorrow)